src/carbon_calculator/carbonCalculator.py

```diff
@@ -114,12 +114,69 @@ def SavePic2Media(picURL):
         print("Error encountered: "+str(e))
         return None
 
-def AverageImpact(action, date=None, locality="default"):
+
+def getCarbonImpact(action, done_only=True): 
+  
+    if not action: return 0
+
+    if hasattr(action, "status"):
```

Comment:
This condition will always pass if `status` is a field on the action model. `hasattr` doesn't check for the value but the key. I don't know if that's what you want.

Reply:
There is probably a better way to do this.  I want to tell whether `action` is a UserActionRel or an Action, so I checked whether the object has a `status` attribute which would indicate it was a UserActionRel.
I first tried importing those models from database.models, but that led to a circular import, since database.models imports the carbon_calculator.models.